src/Raven.Server/Documents/TransactionOperationsMerger.cs

```diff
@@ -60,7 +62,12 @@ public TransactionOperationsMerger(DocumentDatabase parent, CancellationToken sh
 
         public void Start()
         {
-            _txLongRunningOperation = PoolOfThreads.GlobalRavenThreadPool.LongRunning(x => MergeOperationThreadProc(), null, TransactionMergerThreadName);            
+            _txLongRunningOperation = PoolOfThreads.GlobalRavenThreadPool.LongRunning(x => MergeOperationThreadProc(), null, TransactionMergerThreadName);
+        }
+
+        public abstract class DocumentPutTransactionCommand : MergedTransactionCommand
+        {
+            public bool CheckIfGeneratedIdIsNotOverlapping;
```

Comment:
Why, this is relevant only on PUT on new docs. Not relevant anywhere else (`PATCH`, for example)